core/app/models/spree/payment.rb

```diff
@@ -8,9 +8,19 @@ module Spree
   class Payment < Spree::Base
     include Spree::Payment::Processing
 
+    class InvalidStateChange < StandardError; end
+
     alias_attribute :identifier, :number
     deprecate :identifier, :identifier=, deprecator: Spree::Deprecation
 
+    PENDING = 'pending'
+    PROCESSING = 'processing'
+    CHECKOUT = 'checkout'
+    COMPLETED = 'completed'
+    INVALID = 'invalid'
+    VOID = 'void'
+    FAILED = 'failed'
```

Comment:
We should freeze these strings since they shouldn't be mutatable.

Reply:
Will do!

Reply:
Oh, just realised the `frozen_string_literal` at the top of the file takes care of this already.

Reply:
Ah, right. We should delete all of the other freeze calls then :p